packages/hooks/src/types.ts

```diff
@@ -0,0 +1,95 @@
+import { ReactNode } from 'react';
+import {
+  ApolloClient,
+  ApolloQueryResult,
+  ObservableQuery
+} from 'apollo-client';
+import { Observable } from 'apollo-link';
+import {
+  OperationVariables,
+  QueryFunctionOptions,
+  QueryResult,
+  BaseMutationOptions,
+  MutationResult,
+  MutationFunctionOptions,
+  ExecutionResult,
+  BaseSubscriptionOptions,
+  SubscriptionResult
+} from '@apollo/react-common';
+import { DocumentNode } from 'graphql';
+
+/* Common types */
+
+export type CommonOptions<TOptions> = TOptions & {
+  client?: ApolloClient<object>;
+};
+
+/* Query types */
+
+export interface QueryOptions<TData = any, TVariables = OperationVariables>
+  extends QueryFunctionOptions<TData, TVariables> {
+  children?: (result: QueryResult<TData, TVariables>) => ReactNode;
+  query: DocumentNode;
+}
+
+export interface QueryHookOptions<TData = any, TVariables = OperationVariables>
```

Comment:
Might be on purpose, but this does not match the signature in `react-apollo-hooks`: https://github.com/trojanowski/react-apollo-hooks/blob/e590e8f21f1ae4871d641d9681f7932433c8daca/src/useQuery.ts#L35-L45

This means that https://graphql-code-generator.com generates the wrong code. Not sure where the fix should go though 🙂 

/cc @trojanowski

Reply:
link to the issue for that lib: https://github.com/dotansimha/graphql-code-generator/issues/1863

but it seems like they'll swap based on the `importHooksFrom` signature